packages/pdfrx/lib/src/widgets/pdf_page_layout.dart

```diff
@@ -0,0 +1,1094 @@
+///   documentSize: Size(width, height),
+/// );
+/// ```
+///
+/// **Advanced usage (subclassing):**
+/// For dynamic layouts that respond to viewport changes or fit modes:
+/// 1. Extend this class and override [layoutBuilder] for custom positioning logic
+/// 2. Override [primaryAxis] if not vertical scrolling
+/// 3. Override [calculateFitScale] for custom scaling logic
+///
+/// **Document size and margin handling:**
+/// Use [PdfLayoutHelper.widthWithMargins] and [PdfLayoutHelper.heightWithMargins] to properly
+/// include both boundary margins and content margins in your document size calculations.
+/// The helper handles the complexity of margin application so you don't have to.
+///
+/// **Scaling considerations:**
+/// - [calculateFitScale] returns the scale based on [FitMode] strategy
+/// - This is typically used as the minimum scale for the InteractiveViewer,
+///   unless an explicit [PdfViewerParams.minScale] parameter is set
+/// - Override only if default implementation doesn't fit your layout's needs
+class PdfPageLayout {
+  PdfPageLayout({required this.pageLayouts, required this.documentSize})
+    : primaryAxis = documentSize.height >= documentSize.width ? Axis.vertical : Axis.horizontal {
+    _impliedMargin = _calcImpliedMargin();
+  }
+
+  /// Page rectangles positioned within the document coordinate space.
+  ///
+  /// Each rect represents a page's position and size. The rects include positioning
+  /// with spacing between pages, but the rect dimensions themselves are
+  /// the page sizes WITHOUT margins added to width/height.
+  ///
+  /// Use [getPageRectWithMargins] to get a page rect with margins included.
+  final List<Rect> pageLayouts;
+
+  /// Total document size including content margins.
+  ///
+  /// This is the size of the scrollable content area and includes [_impliedMargin] spacing
+  /// on all sides. Does NOT include boundary margins - those are handled separately
+  /// at the viewport level.
+  ///
+  final Size documentSize;
+
+  /// The primary scroll axis for this layout.
+  ///
+  /// In the base class, derived from document dimensions: [Axis.vertical] if
+  /// height >= width, otherwise [Axis.horizontal].
+  ///
+  /// Subclasses can override this to use explicit scroll direction instead.
+  /// For example, [SequentialPagesLayout] overrides to use its scroll direction parameter.
+  ///
+  /// Determines the direction of scrolling and page layout.
+  final Axis primaryAxis;
+
+  /// Content margin around the document edges.
+  ///
+  /// Calculated automatically via [_calcImpliedMargin] based on the spacing
+  /// between page dimensions and document size.
+  ///
+  /// See [getPageRectWithMargins] to get page bounds including margins.
+  late double _impliedMargin;
+
+  /// Get the spread bounds for a given page number.
+  ///
+  /// For single page layouts, this simply returns the page bounds.
+  Rect getSpreadBounds(int pageNumber) {
```

Comment:
TODO: We should re-consider the name of this method since in the base class; "spread" is really a good naming of such page grouping only in specific layouts??